src/GraphQLLocalStrategy.ts

```diff
@@ -1,51 +1,44 @@
+import { Request as ExpressRequest } from 'express';
 /* eslint-disable no-param-reassign */
 import { Strategy as PassportStrategy } from 'passport-strategy';
-import { Request as ExpressRequest } from 'express';
-import { PassportContext, IVerifyOptions } from './types';
-
-type VerifyFn = (username: unknown, password: unknown, done: () => void) => void;
-type VerifyFnWRequest = <U extends {}, Request extends object = ExpressRequest>(
-  req: Request | PassportContext<U, Request>,
-  username: unknown,
-  password: unknown,
-  done: (error: any, user?: any, options?: IVerifyOptions) => void,
-) => void;
+import { IVerifyOptions } from './types';
+
+type VerifyFn<TInput, TUser extends {}, TRequest extends ExpressRequest> = ({
+  req,
+  input,
+  done,
+}: {
+  req: TRequest;
+  input: TInput;
+  done: (error: any, user: TUser, options?: IVerifyOptions) => void;
+}) => void;
 
 interface GraphQLLocalStrategyOptions {
   passReqToCallback?: boolean;
 }
 
-class GraphQLLocalStrategy<U extends {}, Request extends ExpressRequest = ExpressRequest> extends PassportStrategy {
-  constructor(
-    options?: GraphQLLocalStrategyOptions | VerifyFn | VerifyFnWRequest,
-    verify?: VerifyFn | VerifyFnWRequest,
-  ) {
+class GraphQLLocalStrategy<
+  TInput,
+  TUser extends {},
+  TRequest extends ExpressRequest = ExpressRequest
+> extends PassportStrategy {
+  constructor(verify: VerifyFn<TInput, TUser, TRequest>, options?: GraphQLLocalStrategyOptions) {
```

Comment:
This breaks the previous API - do we have a strong reason behind doing this? IT seems like we should overload the constructor instead and keep the original handling?

Reply:
Well, I think this way is much better than having the parameters flipped and with all the weird logic as it was before. But overloading the constructor could also work. I'll try to change it.

Reply:
I agree, although I think that is "classical JS-style". If I remember correctly it has the same pattern in passport itself, e.g. https://github.com/jaredhanson/passport/blob/master/lib/strategies/session.js#L14-L24 - the natural pattern for graphql-passport should be to retain the original style but in a modern TypeScript manner.

Reply:
Also, should I still keep the two functions, one with and one without the request? It would certainly make more sense to just pass an object and the user can choose what they want. But that of course again breaks the API.

Reply:
If we decide to provide an object to the `verify` function instead of separate parameters, that means we would not need the options object, at least not for now, because the request would get passed to the callback every time.